pilot/pkg/model/push_context.go

```diff
@@ -1407,6 +1412,7 @@ func (ps *PushContext) initSidecarScopes(env *Environment) error {
 			if sidecarConfig.Namespace == ps.Mesh.RootNamespace &&
 				sidecarConfig.Spec.(*networking.Sidecar).WorkloadSelector == nil {
 				rootNSConfig = &sidecarConfig
+				ps.defaultSidecarConfig = rootNSConfig
```

Comment:
What happens if there is no `rootNSConfig`?  @howardjohn  has a draft PR for this

Reply:
The same as before:
`GetSidecarScope()` will pass nil to `ConvertToSidecarScope()` which falls back to `DefaultSidecarScopeForNamespace()` which is also called right now.

Looks like it would be wise to close this one though if this is already worked on.